### PR TITLE
New SQL Managed Instance template with Routes and NSGs

### PR DESCRIPTION
**Pull request by jftl6y**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: azure-sql-managed-instance/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+<img src="https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazure.png"/>
+</a>
+
+<a href="https://portal.azure.us/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2Fjftl6y%2Fazure-quickstart-templates%2Fazure-sql-managed-instance%2Fmaster%2Fazuredeploy.json" target="_blank">
+<img src="https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazuregov.png"/>
+</a>
+
+This template deploys an **Azure SQL Managed Instance** into Azure with an correct routes and NSGs per [Connectivity architecture for a managed instance in Azure SQL Database](https://docs.microsoft.com/en-us/azure/sql-database/sql-database-managed-instance-connectivity-architecture).
+
+> [!NOTE]
```
Comment (bmoore-msft) [inline azure-sql-managed-instance/README.md]: this markdown isn't rendering as it seems it should...

File: azure-sql-managed-instance/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+                    },
+                    "miManagementIps": {
+                        "value": "[parameters('miManagementIps')]"
+                    }
+                }
+            }
+        },
+        {
+            "type": "Microsoft.Sql/managedInstances",
+            "apiVersion": "2015-05-01-preview",
```
Comment (bmoore-msft) [inline azure-sql-managed-instance/azuredeploy.json]: update apiVersion

File: azure-sql-managed-instance/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+                "type": "SystemAssigned"
+            },
+            "tags": "[parameters('tags')]",
+            "dependsOn": [
+                "Microsoft.Resources/deployments/BuildMINetworking"
+            ],
+            "properties": {
+                "administratorLogin": "[parameters('sqlManagedInstanceAdminLogin')]",
+                "administratorLoginPassword": "[parameters('sqlManagedInstancePassword')]",
+                "subnetId": "[resourceId(resourceGroup().name,'Microsoft.Network/virtualNetworks/subnets',parameters('vnetResourceName'),parameters('managedInstanceSubnetName'))]",
```
Comment (bmoore-msft) [inline azure-sql-managed-instance/azuredeploy.json]: remove resourceGroup().name

File: azure-sql-managed-instance/README.md (hunk 1 above)
Comment (jftl6y) [reply azure-sql-managed-instance/README.md]: Fixed

File: azure-sql-managed-instance/azuredeploy.json (hunk 3 above)
Comment (jftl6y) [reply azure-sql-managed-instance/azuredeploy.json]: Fixed

File: azure-sql-managed-instance/azuredeploy.json (hunk 2 above)
Comment (jftl6y) [reply azure-sql-managed-instance/azuredeploy.json]: Nevermind - I queried the REST API

**Pull request merged**